src/storage/src/hummock/event_handler/uploader/mod.rs

```diff
@@ -807,7 +826,20 @@ impl UnsyncData {
             .get_mut(&instance_id)
             .expect("should exist");
         let epoch = instance_data.local_seal_epoch(next_epoch);
-        self.epochs.insert(next_epoch, ());
+        // When drop/cancel a streaming job, for the barrier to stop actor, the
+        // local instance will call `local_seal_epoch`, but the `next_epoch` won't be
+        // called `start_epoch` because we have stopped writing on it.
```

Comment:
Does it mean that the actors for the dropping/cancelling streaming job will be included in `actors_to_send` but the table_ids for the streaming job will not be included in the barrier?

Reply:
Yes. This is the key difference between `table_ids_to_sync` and `actors_to_collect` in the `InjectBarrierRequest`. `actors_to_collect` means actors that this barrier will flow through, and the dropping streaming job will stop after the barrier flow through the actor. 

On the other hand, `table_ids_to_sync` means the table id that will start writing data on the `epoch.curr` after the barrier, so it won't include the table_ids of dropping streaming job. For this reason, in `LocalBarrierWorker`, we store the `table_ids_to_sync` of the previous barrier (let's say epoch{curr=2, prev=1}), and when a barrier(epoch{curr=3, prev=2}) flows through the streaming graph, it means the table id in the `table_ids_to_sync` of the previous barrier(epoch{curr=2, prev=1}) has finished writing the data of epoch 2, and then we call `sync` with the `table_ids_to_sync` of the previous barrier.